src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -249,7 +247,10 @@ public T Dequeue()
 
             T removed = _array[_head];
             _array[_head] = default(T);
-            _head = (_head + 1) % _array.Length;
+            if (++_head == _array.Length)
```

Comment:
You could use a local variable to ensure `_head` is only written once by this method. This would preserve the original behavior and is highly unlikely to negatively affect performance.